flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/KafkaSourceBuilder.java

```diff
@@ -396,6 +396,11 @@
                 props);
     }
 
+    /** setParitionDiscoverySetting . */
```

Comment:
This method seems quite confusing. I would suggest the following:
1. Make `parseAndSetRequiredProperties()` package private and annotate it as `VisibleForTesting`.
2. Add a package private `getProps()` method to the `KafkaSourceBuilder` and annotate it as `VisibleForTesting`.

The unit test can just use the above two methods without exposing them to the end users.

Reply:
@becketqin  Thanks  for comment,ACK！VisibleForTesting is better ! i will update it

Reply:
@becketqin, please take a look this updates,thanks